src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -169,39 +185,57 @@ private static IntPtr HostFxrResolver(Assembly assembly, string libraryName)
         private static string? GetSdkFromGlobalSettings(string workingDirectory)
         {
             string? resolvedSdk = null;
-            int rc = NativeMethods.hostfxr_resolve_sdk2(exe_dir: s_dotnetPath.Value, working_dir: workingDirectory, flags: 0, result: (key, value) =>
+            foreach (string dotnetPath in s_dotnetPathCandidates.Value)
             {
-                if (key == NativeMethods.hostfxr_resolve_sdk2_result_key_t.resolved_sdk_dir)
+                int rc = NativeMethods.hostfxr_resolve_sdk2(exe_dir: dotnetPath, working_dir: workingDirectory, flags: 0, result: (key, value) =>
                 {
-                    resolvedSdk = value;
-                }
-            });
+                    if (key == NativeMethods.hostfxr_resolve_sdk2_result_key_t.resolved_sdk_dir)
+                    {
+                        resolvedSdk = value;
+                    }
+                });
 
-            return rc != 0
+                if (rc == 0 && !string.IsNullOrEmpty(resolvedSdk))
+                {
+                    SetEnvironmentVariableIfEmpty("DOTNET_HOST_PATH", dotnetPath);
+                    break;
+                }
+            }
+            
+            return string.IsNullOrEmpty(resolvedSdk)
```

Comment:
Is the difference between this condition and the `(rc == 0 && !string.IsNullOrEmpty(resolvedSdk))` a few lines above intentional? Do we want to handle the case where `hostfxr_resolve_sdk2` returns non-zero but the callbacks sets a non-empty `resolvedSdk`? Maybe the code inside the foreach loop could return instead of breaking.

Reply:
done